source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -533,16 +490,133 @@ namespace AdaptiveCards::Rendering::Uwp
                         m_copyStreamOperations.push_back(copyStreamOperation);
 
                         copyStreamOperation.Completed(
-                            [randomAccessStream](winrt::IAsyncOperationWithProgress<uint64_t, uint64_t> const& /*operation*/,
-                                                 winrt::AsyncStatus /*status*/) { randomAccessStream.Seek(0); });
+                            [randomAccessStream, weakThis, imgProperties, image](
+                                auto const& /*operation*/, auto status)
+                            {
+                                if (status == winrt::AsyncStatus::Completed)
+                                {
+                                    randomAccessStream.Seek(0);
+                                    if (auto strongThis = weakThis.get())
+                                    {
+                                        strongThis->HandleAccessStreamForImageSource(
```

Comment:
see what happens to this one too when the stream is garbage

Reply:
I added a few changes here.

- Catch error if the dataStream is not read properly or if xml string cannot be loaded
- Ensure that the root element is an svg
- Null check for svgString

Let me know what you think!